# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Document.cpp
# Commit: 22f567c00af3
# Full Hash: 22f567c00af321b797590746f181763abb07d83b
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 4. Split FontFaceSetDocumentImpl from FontFaceSetImpl. r=emilio
#   
#   This patch splits document specific functionality from FontFaceSetImpl
#   into a new class FontFaceSetDocumentImpl. This will make it easier to
#   fork FontFaceSetImpl for workers.
# ==============================================================================

diff -r 609a5d93bf08 -r 22f567c00af3 dom/base/Document.cpp
--- a/dom/base/Document.cpp	Tue Jun 28 15:58:10 2022 +0000
+++ b/dom/base/Document.cpp	Tue Jun 28 15:58:11 2022 +0000
@@ -16122,8 +16122,7 @@
     }
 
     if (!mFontFaceSet && !rules.IsEmpty()) {
-      nsCOMPtr<nsPIDOMWindowInner> window = do_QueryInterface(GetScopeObject());
-      mFontFaceSet = new FontFaceSet(window, this);
+      mFontFaceSet = FontFaceSet::CreateForDocument(this);
     }
 
     bool changed = false;
@@ -16155,8 +16154,7 @@
 
 FontFaceSet* Document::Fonts() {
   if (!mFontFaceSet) {
-    nsCOMPtr<nsPIDOMWindowInner> window = do_QueryInterface(GetScopeObject());
-    mFontFaceSet = new FontFaceSet(window, this);
+    mFontFaceSet = FontFaceSet::CreateForDocument(this);
     FlushUserFontSet();
   }
   return mFontFaceSet;